lib/install/phases/coredns.go

```diff
@@ -163,6 +145,34 @@ func (r *corednsExecutor) Rollback(context.Context) error {
 
 // GenerateCorefile will generate a coredns configuration file to be used from within the cluster
 func GenerateCorefile(config CorednsConfig) (string, error) {
+	// Read the resolv.conf from the host doing installation // upgrade
+	// it will be used for configuring coredns upstream servers
+	resolvConf, err := systeminfo.ResolvFromFile("/etc/resolv.conf")
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+
+	// Optionally try and load upstream nameservers from systemd-resolved by reading the compatibility resolv.conf
+	// More Info: https://github.com/gravitational/gravity/issues/606#issuecomment-529171440
+	// TODO(knisbet) is there a better way to pull upstream resolvers directly from systemd?
+	systemdResolvConf, err := systeminfo.ResolvFromFile("/run/systemd/resolve/resolv.conf")
+	if err != nil && !trace.IsNotFound(err) {
+		return "", trace.Wrap(err)
+	}
+
+	config.UpstreamNameservers = mergeUpstreamResolvers(resolvConf, systemdResolvConf)
+	if resolvConf != nil && resolvConf.Rotate {
+		config.Rotate = true
+	}
+	if systemdResolvConf != nil && systemdResolvConf.Rotate {
```

Comment:
Nit: I'd do `else if` here to avoid an extra check in case the above branch already set it to true.